### PR TITLE
Add video tag loaders

### PR DESCRIPTION
**Pull request by notspiff**
This is preparatory work for adding video info tags from containers.

As I am sure there are lots of things to nitpick, I open the PR early.

### CODE REVIEW

**Comment by jumppro** [comment]
Thanks for all you're hard work

**Comment by notspiff** [comment]
more drafting. see commit description - mkv can now hold info, sort of.

somebody needs to come up with the ebml for the tags we want to use on the mkv level, i cba to think that through.

File: xbmc/video/tags/VideoInfoTagLoaderFactory.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                                         bool lookInFolder)
+{
+  // don't try to read tags for streams
+  if (item.IsInternetStream())
+    return nullptr;
+
+  if (item.IsType(".mkv") || item.IsType(".mp4"))
+  {
+    CVideoTagLoaderFFmpeg* ff = new CVideoTagLoaderFFmpeg(item, info, lookInFolder);
+    if (ff->HasInfo());
```
Comment (Rechi) [inline xbmc/video/tags/VideoInfoTagLoaderFactory.cpp]: semicolon at the end of the line looks wrong

File: xbmc/video/tags/VideoTagLoaderFFmpeg.cpp [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+}
+
+bool CVideoTagLoaderFFmpeg::HasInfo() const
+{
+  if (!m_file)
+    return false;
+
+  for (size_t i = 0; i < m_fctx->nb_streams; ++i)
+  {
+    AVDictionaryEntry* avtag;
```
Comment (Rechi) [inline xbmc/video/tags/VideoTagLoaderFFmpeg.cpp]: variables is used uninitialized

File: xbmc/video/tags/VideoInfoTagLoaderFactory.cpp (hunk 1 above)
Comment (notspiff) [reply xbmc/video/tags/VideoInfoTagLoaderFactory.cpp]: indeed.

File: xbmc/video/tags/VideoTagLoaderFFmpeg.cpp (hunk 2 above)
Comment (notspiff) [reply xbmc/video/tags/VideoTagLoaderFFmpeg.cpp]: indeed.

File: xbmc/video/tags/VideoTagLoaderFFmpeg.cpp [hunk 3]
```diff
@@ -157,3 +172,31 @@ CInfoScanner::INFO_TYPE CVideoTagLoaderFFmpeg::Load(CVideoInfoTag& tag, bool)
 
   return CInfoScanner::NO_NFO;
 }
+
+CInfoScanner::INFO_TYPE CVideoTagLoaderFFmpeg::LoadMP4(CVideoInfoTag& tag)
+{
+  AVDictionaryEntry* avtag=nullptr;
```
Comment (xhaggi) [inline xbmc/video/tags/VideoTagLoaderFFmpeg.cpp]: Missing spaces

File: xbmc/video/tags/VideoTagLoaderFFmpeg.cpp [hunk 4]
```diff
@@ -157,3 +172,31 @@ CInfoScanner::INFO_TYPE CVideoTagLoaderFFmpeg::Load(CVideoInfoTag& tag, bool)
...
+  while ((avtag = av_dict_get(m_fctx->metadata, "", avtag, AV_DICT_IGNORE_SUFFIX)))
+  {
+    if (strcmp(avtag->key, "title") == 0)
+      tag.SetTitle(avtag->value);
+    else if (strcmp(avtag->key, "composer") == 0)
+    {
+      std::vector<std::string> writers = StringUtils::Split(avtag->value, " / ");
+      tag.SetWritingCredits(writers);
```
Comment (xhaggi) [inline xbmc/video/tags/VideoTagLoaderFFmpeg.cpp]: Could be `tag.SetWritingCredits(StringUtils::Split(avtag->value, " / "))`

File: xbmc/video/tags/VideoTagLoaderFFmpeg.cpp [hunk 5]
```diff
@@ -157,3 +172,31 @@ CInfoScanner::INFO_TYPE CVideoTagLoaderFFmpeg::Load(CVideoInfoTag& tag, bool)
...
+    }
+    else if (strcmp(avtag->key, "genre") == 0)
+    {
+      std::vector<std::string> genres = StringUtils::Split(avtag->value, " / ");
+      tag.SetGenre(genres);
```
Comment (xhaggi) [inline xbmc/video/tags/VideoTagLoaderFFmpeg.cpp]: Same here

File: xbmc/video/tags/VideoTagLoaderFFmpeg.cpp [hunk 6]
```diff
@@ -176,8 +176,23 @@ CInfoScanner::INFO_TYPE CVideoTagLoaderFFmpeg::LoadMKV(CVideoInfoTag& tag)
...
+  bool hastag = false;
+  while ((avtag = av_dict_get(m_fctx->metadata, "", avtag, AV_DICT_IGNORE_SUFFIX)))
+  {
+    if (strcasecmp(avtag->key, "title") == 0)
+      tag.SetTitle(avtag->value);
+    else if (strcasecmp(avtag->key, "director") == 0)
+    {
+      std::vector<std::string> dirs = StringUtils::Split(avtag->value, " / ");
+      tag.SetDirector(dirs);
+    }
```
Comment (xhaggi) [inline xbmc/video/tags/VideoTagLoaderFFmpeg.cpp]: Same here

File: xbmc/utils/EmbeddedArt.cpp [hunk 7]
```diff
@@ -21,15 +21,17 @@
 #include "EmbeddedArt.h"
 #include "Archive.h"
 
-EmbeddedArtInfo::EmbeddedArtInfo(size_t siz, const std::string &mim)
+EmbeddedArtInfo::EmbeddedArtInfo(size_t siz,
+                                 const std::string &mim, const std::string& t)
```
Comment (xhaggi) [inline xbmc/utils/EmbeddedArt.cpp]: please name it `type` instead of `t` like you did further down

File: xbmc/utils/EmbeddedArt.cpp [hunk 8]
```diff
@@ -21,15 +21,17 @@
...
 {
-  set(siz, mim);
+  set(siz, mim, t);
 }
 
-void EmbeddedArtInfo::set(size_t siz, const std::string &mim)
+void EmbeddedArtInfo::set(size_t siz, const std::string &mim, const std::string& t)
```
Comment (xhaggi) [inline xbmc/utils/EmbeddedArt.cpp]: same here

File: xbmc/utils/EmbeddedArt.h [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  EmbeddedArtInfo(size_t size, const std::string &mime);
+  ~EmbeddedArtInfo() override = default;
+
+  // implementation of IArchivable
+  void Archive(CArchive& ar) override;
+
+  void set(size_t size, const std::string &mime);
+  void clear();
+  bool empty() const;
+  bool matches(const EmbeddedArtInfo &right) const;
```
Comment (xhaggi) [inline xbmc/utils/EmbeddedArt.h]: I know it's not your fault, but if you had some time, could use pls rename the methods to upper-case

File: xbmc/video/VideoThumbLoader.cpp [hunk 10]
```diff
@@ -564,6 +564,17 @@ std::string CVideoThumbLoader::GetEmbeddedThumbURL(const CFileItem &item)
...
+/*bool CVideoThumbLoader::GetEmbeddedThumb(const std::string& path, EmbeddedArt& art)
+{
+  CFileItem item(path, false);
+  std::unique_ptr<IVideoInfoTagLoader> pLoader (CVideoInfoTagLoaderFactory::CreateLoader(item));
+  CVideoInfoTag tag;
+  if (null != pLoader.get())
+    pLoader->Load(tag, art, false);
+
+  return !art.empty();
+}*/
```
Comment (xhaggi) [inline xbmc/video/VideoThumbLoader.cpp]: left-over?

File: xbmc/video/tags/VideoTagLoaderFFmpeg.cpp [hunk 11]
```diff
@@ -118,6 +118,13 @@ bool CVideoTagLoaderFFmpeg::HasInfo() const
   AVDictionaryEntry* avtag=nullptr;
   while ((avtag = av_dict_get(m_fctx->metadata, "", avtag, AV_DICT_IGNORE_SUFFIX)))
     std::cout << avtag->key << ": " << avtag->value << std::endl;
+  std::cout << "---------" << std::endl;
+  for (size_t i = 0; i < m_fctx->nb_streams; ++i)
+  {
+    AVDictionaryEntry* avtag=nullptr;
+    while ((avtag = av_dict_get(m_fctx->streams[i]->metadata, "", avtag, AV_DICT_IGNORE_SUFFIX)))
+      std::cout << avtag->key << ": " << avtag->value << std::endl;
```
Comment (xhaggi) [inline xbmc/video/tags/VideoTagLoaderFFmpeg.cpp]: `std::cout` should be gone, right?

File: xbmc/video/tags/VideoTagLoaderFFmpeg.cpp [hunk 12]
```diff
@@ -132,8 +139,44 @@ bool CVideoTagLoaderFFmpeg::HasInfo() const
...
+  else
+    return CInfoScanner::NO_NFO;
+
+}
+
+CInfoScanner::INFO_TYPE CVideoTagLoaderFFmpeg::LoadMKV(CVideoInfoTag& tag,
+                                                       std::vector<EmbeddedArt>* art)
 {
+  // embedded art
+  for (size_t i = 0; i < m_fctx->nb_streams; ++i)
```
Comment (xhaggi) [inline xbmc/video/tags/VideoTagLoaderFFmpeg.cpp]: range based loop?

File: xbmc/video/tags/VideoTagLoaderFFmpeg.cpp [hunk 13]
```diff
@@ -132,8 +139,44 @@ bool CVideoTagLoaderFFmpeg::HasInfo() const
...
+  {
+    AVDictionaryEntry* avtag=nullptr;
```
Comment (xhaggi) [inline xbmc/video/tags/VideoTagLoaderFFmpeg.cpp]: missing spaces

File: xbmc/TextureCacheJob.cpp [hunk 14]
```diff
@@ -198,6 +201,14 @@ CBaseTexture *CTextureCacheJob::LoadImage(const std::string &image, unsigned int
       return CBaseTexture::LoadFromFileInMemory(&art.data[0], art.size, art.mime, width, height);
   }
 
+  size_t tpos = additional_info.find("video_");
+  if (tpos != std::string::npos)
```
Comment (xhaggi) [inline xbmc/TextureCacheJob.cpp]: `if (StringUtils::StartsWith(additional_info, "video_")`

File: xbmc/TextureCacheJob.cpp [hunk 15]
```diff
@@ -167,6 +168,8 @@ std::string CTextureCacheJob::DecodeImageURL(const std::string &url, unsigned in
       return "";
     if (thumbURL.GetUserName() == "music")
       additional_info = "music";
+    if (thumbURL.GetUserName().find("video_") != std::string::npos)
```
Comment (xhaggi) [inline xbmc/TextureCacheJob.cpp]: `StringUtils::StartsWith(thumbURL.GetUserName(), "video_")`

File: xbmc/video/VideoInfoScanner.cpp [hunk 16]
```diff
@@ -1408,6 +1408,19 @@ namespace VIDEO
...
     }
 
+    // find embedded art
+    if (pItem->HasVideoInfoTag() && !pItem->GetVideoInfoTag()->m_coverArt.empty())
+    {
+      for (auto& it : pItem->GetVideoInfoTag()->m_coverArt)
+      {
+        if (art.find(it.type) == art.end())
+        {
+          std::string thumb = CTextureUtils::GetWrappedImageURL(pItem->GetPath(), "video_"+it.type);
```
Comment (xhaggi) [inline xbmc/video/VideoInfoScanner.cpp]: missing spaces before and after `+`

File: xbmc/video/VideoThumbLoader.cpp [hunk 17]
```diff
@@ -358,6 +360,21 @@ bool CVideoThumbLoader::LoadItemLookup(CFileItem* pItem)
...
+      else
+      {
+        // If nothing was found, try embedded art
+        if (pItem->HasVideoInfoTag() && !pItem->GetVideoInfoTag()->m_coverArt.empty())
+        {
+          for (auto& it : pItem->GetVideoInfoTag()->m_coverArt)
+          {
+            if (it.type == type)
+            {
+              art = CTextureUtils::GetWrappedImageURL(pItem->GetPath(), "video_"+type);
```
Comment (xhaggi) [inline xbmc/video/VideoThumbLoader.cpp]: same here

File: xbmc/video/VideoThumbLoader.cpp [hunk 18]
```diff
@@ -513,6 +530,22 @@ bool CVideoThumbLoader::FillThumb(CFileItem &item)
...
+  else
+  {
+    // If nothing was found, try embedded art
+    if (item.HasVideoInfoTag() && !item.GetVideoInfoTag()->m_coverArt.empty())
+    {
+      for (auto& it : item.GetVideoInfoTag()->m_coverArt)
+      {
+        if (it.type == "thumb")
+        {
+          thumb = CTextureUtils::GetWrappedImageURL(item.GetPath(), "video_"+it.type);
```
Comment (xhaggi) [inline xbmc/video/VideoThumbLoader.cpp]: same here

File: xbmc/TextureCache.cpp [hunk 19]
```diff
@@ -100,7 +100,8 @@ std::string CTextureCache::GetCachedImage(const std::string &image, CTextureDeta
 
 bool CTextureCache::CanCacheImageURL(const CURL &url)
 {
-  return (url.GetUserName().empty() || url.GetUserName() == "music");
+  return url.GetUserName().empty() || url.GetUserName() == "music" ||
+          StringUtils::StartsWith(url.GetUserName(), "video_");
```
Comment (garbear) [inline xbmc/TextureCache.cpp]: typo on `video_`?

File: xbmc/InfoScanner.h [hunk 20]
```diff
@@ -41,6 +41,20 @@ class CInfoScanner
     INFO_ADDED
   };
 
+  /*
+   \brief Type of information returned from tag readers.
+  */
+
+  enum INFO_TYPE
```
[the remark this answers is not in the document]
Comment (notspiff) [reply xbmc/InfoScanner.h]: we do not need anything but int as underlying type so i don't get why?

File: xbmc/NfoFile.cpp [hunk 21]
```diff
@@ -192,3 +159,43 @@ void CNfoFile::Close()
...
+
+  // add selected scraper - first priority
+  if (selectedScraper)
+    vecScrapers.push_back(selectedScraper);
+
+  // Add all scrapers except selected and default
+  VECADDONS addons;
+  CServiceBroker::GetAddonMgr().GetAddons(addons, type);
+
+  for (unsigned i = 0; i < addons.size(); ++i)
```
[the remark this answers is not in the document]
Comment (notspiff) [reply xbmc/NfoFile.cpp]: done

File: xbmc/NfoFile.cpp [hunk 22]
```diff
@@ -192,3 +159,43 @@ void CNfoFile::Close()
...
+  {
+    ScraperPtr scraper = std::dynamic_pointer_cast<CScraper>(addons[i]);
+
+    // skip if scraper requires settings and there's nothing set yet
+    if (scraper->RequiresSettings() && !scraper->HasUserSettings())
+      continue;
+
+    if( (!selectedScraper || selectedScraper->ID() != scraper->ID())
+        && (!defaultScraper || defaultScraper->ID() != scraper->ID()) )
```
[the remark this answers is not in the document]
Comment (notspiff) [reply xbmc/NfoFile.cpp]: done

File: xbmc/NfoFile.cpp [hunk 23]
```diff
@@ -192,3 +159,43 @@ void CNfoFile::Close()
...
+      vecScrapers.push_back(scraper);
+  }
+
+  // add default scraper - not user selectable so it's last priority
+  if( defaultScraper && (!selectedScraper ||
```
[the remark this answers is not in the document]
Comment (notspiff) [reply xbmc/NfoFile.cpp]: done

File: xbmc/TextureCacheJob.cpp (hunk 15 above)
Comment (notspiff) [reply xbmc/TextureCacheJob.cpp]: done.

File: xbmc/TextureCacheJob.cpp (hunk 14 above)
Comment (notspiff) [reply xbmc/TextureCacheJob.cpp]: done

File: xbmc/utils/EmbeddedArt.cpp (hunk 7 above)
Comment (notspiff) [reply xbmc/utils/EmbeddedArt.cpp]: then i would have to rename the member first.

File: xbmc/utils/EmbeddedArt.cpp (hunk 8 above)
Comment (notspiff) [reply xbmc/utils/EmbeddedArt.cpp]: and same answer ;)

File: xbmc/utils/EmbeddedArt.h (hunk 9 above)
Comment (notspiff) [reply xbmc/utils/EmbeddedArt.h]: sure. i don't care about who did what, i want the code more readable for joe random. and as i'm almost blind to cosmetics these days as i write code all over with all sorts of  weird requirements i need help with spotting such things ;) i will do it elsewhere though as i need to split changes and move to keep git history.

File: xbmc/video/VideoInfoScanner.cpp (hunk 16 above)
Comment (notspiff) [reply xbmc/video/VideoInfoScanner.cpp]: done.

File: xbmc/video/VideoThumbLoader.cpp (hunk 10 above)
Comment (notspiff) [reply xbmc/video/VideoThumbLoader.cpp]: outdated commit.

File: xbmc/video/VideoThumbLoader.cpp (hunk 17 above)
Comment (notspiff) [reply xbmc/video/VideoThumbLoader.cpp]: done.

File: xbmc/video/VideoThumbLoader.cpp (hunk 18 above)
Comment (notspiff) [reply xbmc/video/VideoThumbLoader.cpp]: done.

File: xbmc/video/tags/IVideoInfoTagLoader.h [hunk 24]
```diff
@@ -0,0 +1,67 @@
+#pragma once
+/*
+ *      Copyright (C) 2005-2013 Team XBMC
```
[the remark this answers is not in the document]
Comment (notspiff) [reply xbmc/video/tags/IVideoInfoTagLoader.h]: done.

File: xbmc/video/tags/VideoInfoTagLoaderFactory.cpp [hunk 25]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+#include "VideoInfoTagLoaderFactory.h"
+#include "VideoTagLoaderNFO.h"
+#include "addons/AudioDecoder.h"
+
+using namespace VIDEO;
+
+IVideoInfoTagLoader*
+CVideoInfoTagLoaderFactory::CreateLoader(const CFileItem& item,
```
[the remark this answers is not in the document]
Comment (notspiff) [reply xbmc/video/tags/VideoInfoTagLoaderFactory.cpp]: it was to keep it to 80 chars, but it only overflows with a few so done.

File: xbmc/video/tags/VideoInfoTagLoaderFactory.h [hunk 26]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace VIDEO
+{
+  class CVideoInfoTagLoaderFactory
+  {
+  public:
+    //! \brief Returns a tag loader for the given item.
+    //! \param item The item to find tag loader for
+    //! \param type Type of tag loader. In particular used for tvshows
+    static IVideoInfoTagLoader* CreateLoader(const CFileItem& item,
```
[the remark this answers is not in the document]
Comment (notspiff) [reply xbmc/video/tags/VideoInfoTagLoaderFactory.h]: then i would have to create a new item so i can point to it in some caller spots so i don't think so.

File: xbmc/video/tags/VideoTagLoaderFFmpeg.cpp (hunk 11 above)
Comment (notspiff) [reply xbmc/video/tags/VideoTagLoaderFFmpeg.cpp]: debugging code that i keep for the time being. not meant for merge.

File: xbmc/video/tags/VideoTagLoaderFFmpeg.cpp (hunk 12 above)
Comment (notspiff) [reply xbmc/video/tags/VideoTagLoaderFFmpeg.cpp]: can't range over a C array.

File: xbmc/video/tags/VideoTagLoaderFFmpeg.cpp (hunk 13 above)
Comment (notspiff) [reply xbmc/video/tags/VideoTagLoaderFFmpeg.cpp]: superfluous assignment so addressed that instead.

File: xbmc/video/tags/VideoTagLoaderFFmpeg.cpp (hunk 6 above)
Comment (notspiff) [reply xbmc/video/tags/VideoTagLoaderFFmpeg.cpp]: done.

File: xbmc/video/tags/VideoTagLoaderFFmpeg.cpp (hunk 3 above)
Comment (notspiff) [reply xbmc/video/tags/VideoTagLoaderFFmpeg.cpp]: done.

File: xbmc/video/tags/VideoTagLoaderFFmpeg.cpp (hunk 4 above)
Comment (notspiff) [reply xbmc/video/tags/VideoTagLoaderFFmpeg.cpp]: done.

File: xbmc/video/tags/VideoTagLoaderFFmpeg.cpp (hunk 5 above)
Comment (notspiff) [reply xbmc/video/tags/VideoTagLoaderFFmpeg.cpp]: done.

File: xbmc/video/tags/VideoTagLoaderFFmpeg.cpp [hunk 27]
```diff
@@ -0,0 +1,159 @@
...
+    AVDictionaryEntry* avtag;
+    avtag = av_dict_get(m_fctx->streams[i]->metadata, "filename", avtag, AV_DICT_IGNORE_SUFFIX);
+    if (avtag && std::string(avtag->value) == "kodi-metadata")
+    {
+      m_metadata_stream = i;
+      return true;
+    }
+  }
+
+  AVDictionaryEntry* avtag=nullptr;
```
[the remark this answers is not in the document]
Comment (notspiff) [reply xbmc/video/tags/VideoTagLoaderFFmpeg.cpp]: debug code but done anyways.

File: xbmc/TextureCache.cpp (hunk 19 above)
Comment (notspiff) [reply xbmc/TextureCache.cpp]: no, it's image://video_[fanart|poster|thumb|etc]@file

File: xbmc/InfoScanner.h (hunk 20 above)
[the remark this answers is not in the document]
Comment (xhaggi) [reply xbmc/InfoScanner.h]: Do we really need the underlying int value? If not a enum class would be appreciated.

File: xbmc/video/tags/VideoInfoTagLoaderFactory.h (hunk 26 above)
[the remark this answers is not in the document]
Comment (xhaggi) [reply xbmc/video/tags/VideoInfoTagLoaderFactory.h]: Okay

File: xbmc/utils/EmbeddedArt.cpp (hunk 7 above)
Comment (garbear) [reply xbmc/utils/EmbeddedArt.cpp]: succinct variable names are OK if they're documented in the header

File: xbmc/video/tags/VideoTagLoaderNFO.cpp [hunk 28]
```diff
@@ -0,0 +1,210 @@
...
+}
+
+bool CVideoTagLoaderNFO::HasInfo() const
+{
+  return !m_path.empty() && CFile::Exists(m_path);
+}
+
+CInfoScanner::INFO_TYPE CVideoTagLoaderNFO::Load(CVideoInfoTag& tag,
+                                                 bool prioritise,
+                                                 std::vector<EmbeddedArt>*)
```
Comment (garbear) [inline xbmc/video/tags/VideoTagLoaderNFO.cpp]: Missing `art` parameter name (and below too)

Comment (notspiff) [reply xbmc/video/tags/VideoTagLoaderNFO.cpp]: It is called parameter anonymization and is done on purpose.

Comment (garbear) [reply xbmc/video/tags/VideoTagLoaderNFO.cpp]: cool

File: xbmc/InfoScanner.h (hunk 20 above)
[the remark this answers is not in the document]
Comment (notspiff) [reply xbmc/InfoScanner.h]: i don't understand what problem you are trying to resolve by using enum class. it's already scoped by being a class member and we do not need to specify a complex underlying type.

[the remark this answers is not in the document]
Comment (xhaggi) [reply xbmc/InfoScanner.h]: Ok leave it as it is. Don't want to slow down the merge.

[the remark this answers is not in the document]
Comment (notspiff) [reply xbmc/InfoScanner.h]: if you have a point, it's fine. but i can't see it. in any case the merge doesn't ride on his, it rides on the mkv tag spec. the stuff from matroska.org doesn't cut it. at all.

[the remark this answers is not in the document]
Comment (xhaggi) [reply xbmc/InfoScanner.h]: There are plenty of pages explaining why enum classes should be preferred over plain enums.

* They don't convert implicitly to int.
* They don't pollute the surrounding namespace.
* They can be forward-declared.

If we don't need the implicit conversion in that case, why not jump over? It's the same for range-based loops etc. .. the old for-loop over an iterator still works, but we prefer the new one ;)

But please don't change it if you think it's not worth it.

[the remark this answers is not in the document]
Comment (notspiff) [reply xbmc/InfoScanner.h]: i know their merits, it's not about that, but:
- we do not pollute any namespace as it's a class member
- no need for forwarding - i need the underlying class definition in all spots where it is in use.
- no int casts apply as i use them strictly to return status from functions

as far as i see it there is absolutely no gain here, and there is a cost in terms of extra qualification in the code using them.

**Comment by wsnipex** [comment]
this breaks building with taglib 1.9 :(

**Comment by Rechi** [comment]
#13223 should fix it

**Comment by Memphiz** [comment]
@notspiff this PR results in the issue this user reports here right?

https://forum.kodi.tv/showthread.php?tid=326340

**Comment by notspiff** [comment]
If u want to consider it an issue then yes i guess.

**Comment by Memphiz** [comment]
Not really - it's pebcak if you ask me

**Comment by notspiff** [comment]
well, i'm not too concerned about placing blame as such. A new feature was added (in fact on the request of the itunes crowd..), and some get bitten. but it's not an issue, as such, it just.. is ;)

i fully expect that we need to add some sort of disable for tags or something, but i kinda want to wait it out and see how bad it becomes before considering this.

**Pull request merged**